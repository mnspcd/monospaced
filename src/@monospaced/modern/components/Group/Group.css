.mdrn .Group {
  overflow: hidden;
}

.mdrn .Group-inner {
  display: flex;
  flex-wrap: wrap;
  margin-bottom: -0.75em;
  margin-right: -0.75em;
}

.mdrn .Group-itemInner {
  margin-bottom: 0.75em;
  margin-right: 0.75em;
}

.mdrn .Group--tessellate .Group-item,
.mdrn .Group--tessellate .Group-inner::before,
.mdrn .Group--tessellate .Group-inner::after {
  flex: 1 0 25%;
  min-width: calc(150rem / 16);
}

.mdrn .Group--tessellate .Group-inner::before,
.mdrn .Group--tessellate .Group-inner::after {
  content: "";
}

.mdrn .Group--tessellate .Group-inner::before {
  order: 1;
}
