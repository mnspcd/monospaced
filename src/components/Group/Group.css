.Group {
  overflow: hidden;
}

.Group-inner {
  display: flex;
  flex-wrap: wrap;
  margin-bottom: -0.75em;
  margin-right: -0.75em;
}

.Group-itemInner {
  margin-bottom: 0.75em;
  margin-right: 0.75em;
}

.Group--tessellate .Group-item,
.Group--tessellate .Group-inner::before,
.Group--tessellate .Group-inner::after {
  flex: 1 0 25%;
  min-width: calc(150rem / 16);
}

.Group--tessellate .Group-inner::before,
.Group--tessellate .Group-inner::after {
  content: "";
}

.Group--tessellate .Group-inner::before {
  order: 1;
}
