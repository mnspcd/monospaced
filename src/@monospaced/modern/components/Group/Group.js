import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import "./Group.css";

/**
 * Use `Group` to layout content in flexible rows.
 */
const Group = ({ children, tessellate = false }) => (
  <div
    className={classNames({
      Group: true,
      [`Group--tessellate`]: tessellate,
    })}
  >
    <div className="Group-inner">{children}</div>
  </div>
);

Group.propTypes = {
  /**
   * Group.Item components to display in the Group.
   */
  children: PropTypes.node.isRequired,

  /**
   * Should the Group.Items fill their container?
   */
  tessellate: PropTypes.bool,
};

const Item = ({ children }) => (
  <div className="Group-item">
    <div className="Group-itemInner">{children}</div>
  </div>
);

Item.displayName = "Group.Item";

Item.propTypes = {
  /**
   * Content to display in the Group.Item.
   */
  children: PropTypes.node.isRequired,
};

Group.Item = Item;

export { Item };
export default Group;
