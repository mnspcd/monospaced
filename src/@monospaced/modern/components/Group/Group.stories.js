import { boolean, number } from "@storybook/addon-knobs";
import React from "react";

import Block from "../Block";

import Group, { Item } from "./Group";

export default {
  component: Group,
  parameters: {
    subcomponents: {
      "Group.Item": Item,
    },
  },
  title: "Components/Group",
};

export const basic = () => (
  <Group>
    <Group.Item>
      <Block>Item</Block>
    </Group.Item>
    <Group.Item>
      <Block>Item</Block>
    </Group.Item>
    <Group.Item>
      <Block>Item</Block>
    </Group.Item>
    <Group.Item>
      <Block>Item</Block>
    </Group.Item>
  </Group>
);

export const tessellate = () => (
  <Group tessellate>
    <Group.Item>
      <Block>Item</Block>
    </Group.Item>
    <Group.Item>
      <Block>Item</Block>
    </Group.Item>
    <Group.Item>
      <Block>Item</Block>
    </Group.Item>
    <Group.Item>
      <Block>Item</Block>
    </Group.Item>
  </Group>
);

export const knobs = () => {
  const numberOfItems = number(
    "number of items",
    4,
    {
      range: true,
      min: 1,
      max: 16,
    },
    "Story",
  );

  return (
    <Group tessellate={boolean("tessellate", false, "Group")}>
      {[...Array(numberOfItems).keys()].map(i => (
        <Group.Item key={i}>
          <Block>Item</Block>
        </Group.Item>
      ))}
    </Group>
  );
};

knobs.story = { parameters: { docs: { disable: true } } };
