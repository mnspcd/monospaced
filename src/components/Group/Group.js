import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import "./Group.css";

const Group = ({ children, tessellate }) => (
  <div
    className={classNames({
      Group: true,
      [`Group--tessellate`]: tessellate,
    })}
  >
    <div className="Group-inner">{children}</div>
  </div>
);

Group.propTypes = {
  children: PropTypes.node.isRequired,
  tessellate: PropTypes.bool,
};

Group.Item = ({ children }) => (
  <div className="Group-item">
    <div className="Group-itemInner">{children}</div>
  </div>
);

Group.Item.displayName = "Group.Item";

Group.Item.propTypes = { children: PropTypes.node.isRequired };

export default Group;
