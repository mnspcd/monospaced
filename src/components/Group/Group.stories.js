import React from "react";
import { boolean, number } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Group from "./Group";
import Block from "../Block";

storiesOf("Components/Group", module).add("Component", () => {
  const numberOfItems = number(
    "number of items",
    4,
    {
      range: true,
      min: 1,
      max: 16,
    },
    "Story",
  );

  return (
    <Group tessellate={boolean("tessellate", false, "Group")}>
      {[...Array(numberOfItems).keys()].map(i => (
        <Group.Item key={i}>
          <Block>Item</Block>
        </Group.Item>
      ))}
    </Group>
  );
});
