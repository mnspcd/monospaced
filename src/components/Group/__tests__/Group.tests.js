/* eslint-env jest */
import React from "react";
import { render } from "enzyme";

import Group from "../Group";

describe("Group component", () => {
  it("should render correctly", () => {
    const component = render(
      <Group>
        <Group.Item>children</Group.Item>
      </Group>,
    );
    expect(component).toMatchSnapshot();
  });
});
