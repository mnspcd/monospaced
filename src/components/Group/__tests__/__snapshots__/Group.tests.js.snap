// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group component should render correctly 1`] = `
<div
  class="Group"
>
  <div
    class="Group-inner"
  >
    <div
      class="Group-item"
    >
      <div
        class="Group-itemInner"
      >
        children
      </div>
    </div>
  </div>
</div>
`;
